src/framework/uicomponents/qml/MuseScore/UiComponents/Dropdown.qml

```diff
@@ -0,0 +1,224 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQml.Models 2.2
+
+import MuseScore.Ui 1.0
+
+import "internal"
+
+DropdownItem {
+
+    id: root
+
+    property var model: null
+    property alias count: view.count
+
+    property int currentIndex: 0
+    property string currentText: "--"
+    property string currentValue: ""
+
+    property string displayText: root.currentText
+
+    property int popupWidth: root.width
+    property int popupItemsCount: root.defaultPopupItemsCount()
+
+    property alias navigation: navCtrl
+
+    property alias dropIcon: dropIconItem
+
+    property string textRole: "text"
+    property string valueRole: "value"
+
+    text: root.displayText
```

Comment:
Is there any particular reason to have 3 properties like "displayedText", "currentText" and "text" simultaneously? It's a bit confusing which one to use

Reply:
hmm, I will remove displayedText